packages/odie-client/src/components/send-message-input/use-attachment-handler.tsx

```diff
@@ -0,0 +1,166 @@
+		mediaUrl: URL.createObjectURL( file ),
+	} );
+};
+
+export const useAttachmentHandler = () => {
+	const { trackEvent, chat, addMessage, isUserEligibleForPaidSupport } = useOdieAssistantContext();
+
+	const { data: authData } = useAuthenticateZendeskMessaging(
+		isUserEligibleForPaidSupport,
+		'messenger'
+	);
+
+	const { zendeskClientId, connectionStatus } = useSelect( ( select ) => {
+		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+		const connectionStatus = helpCenterSelect.getZendeskConnectionStatus();
+		return {
+			zendeskClientId: helpCenterSelect.getZendeskClientId(),
+			connectionStatus,
+		};
+	}, [] );
+
+	const inferredClientId = chat.clientId ? chat.clientId : zendeskClientId;
+
+	const { isPending: isAttachingFile, mutateAsync: attachFileToConversation } =
+		useAttachFileToConversation();
+
+	const handleFileUpload = useCallback(
+		async ( file: File ) => {
+			if ( file.type.startsWith( 'image/' ) ) {
+				if ( authData && chat.conversationId && inferredClientId && file ) {
+					attachFileToConversation( {
+						authData,
+						file,
+						conversationId: chat.conversationId,
+						clientId: inferredClientId,
+					} ).then( () => {
+						addMessage( getPlaceholderAttachmentMessage( file ) );
+						trackEvent( 'send_message_attachment', { type: file.type } );
+					} );
+				}
+			} else {
+				addMessage( getOdieWrongFileTypeMessage() );
+			}
+		},
+		[
+			authData,
+			chat.conversationId,
+			inferredClientId,
+			attachFileToConversation,
+			addMessage,
+			trackEvent,
+		]
+	);
+
+	const onFilesDrop = ( files: File[] ) => {
+		const file = files?.[ 0 ];
+		if ( file ) {
+			handleFileUpload( file );
+		}
+	};
+
+	const showAttachmentButton = chat.conversationId && inferredClientId;
+	const isLiveChat = chat.provider?.startsWith( 'zendesk' );
+
+	// Handle paste events for images
+	const handleImagePaste = useCallback(
+		( e: React.KeyboardEvent< HTMLTextAreaElement > ) => {
+			if ( e.key === 'v' && ( e.ctrlKey || e.metaKey ) ) {
+				setTimeout( () => {
+					navigator.clipboard
+						.read()
+						.then( ( items ) => {
+							for ( const item of items ) {
+								for ( const type of item.types ) {
+									if ( type.startsWith( 'image/' ) ) {
+										item.getType( type ).then( ( blob ) => {
+											const file = new File( [ blob ], 'pasted-image.png', { type } );
+											handleFileUpload( file );
+										} );
+										break;
+									}
+								}
+							}
+						} )
+						.catch( () => {
+							// Clipboard API not supported
+						} );
+				}, 0 );
+			}
+		},
+		[ handleFileUpload ]
+	);
+
+	const AttachmentDropZone = () => {
+		if ( ! showAttachmentButton ) {
+			return null;
+		}
+
+		return (
+			<DropZone
+				onFilesDrop={ onFilesDrop }
+				label={ __( 'Share this image with our Happiness Engineers', __i18n_text_domain__ ) }
+			/>
+		);
+	};
+
+	const attachmentAction = {
+		id: 'attachment',
+		icon: image,
```

Comment:
The icon is 4.5px too big. (22.5 vs 18px) 

```suggestion
		icon: <Icon size="18" icon={ image } />,
```